openedx/core/djangoapps/dark_lang/middleware.py

```diff
@@ -91,16 +92,32 @@ def process_request(self, request):
             return
 
         self._clean_accept_headers(request)
-        self._set_site_or_microsite_language(request)
-        self._activate_preview_language(request)
 
-    def _set_site_or_microsite_language(self, request):
+    def process_response(self, request, response):
+        """
+        Prevent user from requesting un-released languages except by using the preview-lang query string.
+        """
+        if not DarkLangConfig.current().enabled:
+            return response
+        self._set_site_or_microsite_language(request, response)
+        self._activate_preview_language(request, response)
+
+        return response
+
+    def _set_site_or_microsite_language(self, request, response):
         """
         Apply language specified in site configuration.
         """
         language = get_value('LANGUAGE_CODE', None)
         if language:
             request.session[LANGUAGE_SESSION_KEY] = language
+            response.set_cookie(
+                settings.LANGUAGE_COOKIE_NAME,
+                value=language,
+                domain=settings.SESSION_COOKIE_DOMAIN,
```

Comment:
[question] Should this be `SHARED_COOKIE_DOMAIN`? Only asking because in another place that's what's being used for this cookie. (I don't actually know which is correct!)

Reply:
I used this as a reference and it is setting the domain to `SESSION_COOKIE_DOMAIN`. 

https://github.com/edx/edx-platform/blob/f01ea7b347862d5edfad9832b9e58d57682dc323/openedx/core/djangoapps/lang_pref/middleware.py#L85-L91

But you are right on this place the domain is `SHARED_COOKIE_DOMAIN`. 
https://github.com/edx/edx-platform/blob/f01ea7b347862d5edfad9832b9e58d57682dc323/openedx/core/djangoapps/lang_pref/views.py#L27-L33

Which domain would be correct from the two?

Reply:
I'm assuming this cookie is also intended to control language preference on other IDAs. If so, then we should use `SHARED_COOKIE_DOMAIN` (which will be `edx.org` for us). If it's just for LMS, then `SESSION_COOKIE_DOMAIN`.

Reply:
@jmbowman can you please guide me, which domain should be appropriate for this cookie in this middleware?

Reply:
After some more research, it looks like this should be using the `SHARED_COOKIE_DOMAIN`. Copying from my comment in TNL-8754, about accounts.edx.org not showing the correct language in the language preferences dropdown:

The `prod-edx-language-preference` cookie is currently being set on `courses.edx.org` instead of `edx.org`, and I think that relates to some prep work that was done for the LMS session cookie work: Previously the LMS was setting all cookies with `domain=SESSION_COOKIE_DOMAIN` (which was `edx.org`), but we needed to change `SESSION_COOKIE_DOMAIN` to `courses.edx.org` without mucking with the other cookies.) Arch-BOM introduced `SHARED_COOKIE_DOMAIN: edx.org` and moved all the other cookies to use that setting instead.

This was supposed to include the language cookie but I think we missed a few spots, such as the files touched in this PR.

So I think if you change this to `SHARED_COOKIE_DOMAIN` you'll actually fix a bug we introduced. :-)

Reply:
Replaced the domain with `SHARED_COOKIE_DOMAIN`. And yeah just noticed the accounts language selection bug, hope this change fixes that.